apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -138,7 +133,60 @@ export const Lti13ClaimSchema = z.object({
   // https://www.imsglobal.org/spec/lti/v1p3#vendor-specific-extension-claims
   // My development Canvas sends their own named extension as a top level property
   // "https://www.instructure.com/placement": "course_navigation"
+
+  // https://www.imsglobal.org/spec/lti-ags/v2p0#assignment-and-grade-service-claim
+  'https://purl.imsglobal.org/spec/lti-ags/claim/endpoint': z
+    .object({
+      lineitems: z.string().optional(),
+      lineitem: z.string().optional(),
+      scope: z.string().array(),
+    })
+    .optional(),
+
+  // https://www.imsglobal.org/spec/lti-nrps/v2p0/#resource-link-membership-service
+  'https://purl.imsglobal.org/spec/lti-nrps/claim/namesroleservice': z
+    .object({
+      context_memberships_url: z.string(),
+      service_versions: z.literal('2.0').array(),
+    })
+    .optional(),
```

Comment:
Do these come along for all claims, or just `LtiResourceLinkRequest`? I'm wondering if we can/should push these down into one of the more specific types.

I tried reading the spec, but IMO it was too vague to answer my question. From https://www.imsglobal.org/spec/lti-ags/v2p0/#assignment-and-grade-service-claim:

> This specification introduces the claim https://purl.imsglobal.org/spec/lti-ags/claim/endpoint. This claim MUST be included in LTI messages if any of the Assignment and Grade Services are accessible by the tool in the context of the LTI message.

I don't know what "accessible by the tool in the context of the LTI message" means in the context of `LtiDeepLinkingRequest`.

Reply:
For Canvas, the AGS and NRPS claims come with both types. I guess they're erroring on the side of caution.